src/services/iamRole/connections.ts

```diff
@@ -229,6 +231,30 @@ export default ({
     }
   }
 
+  /**
+   * Find any elasticBeanstalkApp related data
+   */
+  const elasticBApps = data.find(
+    ({ name }) => name === services.elasticBeanstalkApp
+  )
+  if (elasticBApps?.data?.[region]) {
+    const dataAtRegion: RawAwsElasticBeanstalkApp[] = elasticBApps.data[
+      region
+    ].filter(
+      ({
+        ResourceLifecycleConfig: { ServiceRole: iamServiceRole } = {},
+      }: RawAwsElasticBeanstalkApp) => iamServiceRole === role.Arn
```

Comment:
so If the iam service role is the same, it means the elastic beanstalk app and env are related?

Reply:
and wouldnt the id be the id of the elastic beanstalk app not the serviceRole?

Reply:
Answering the first question, the elastic beanstalk app lives in an elastic beanstalk environment. Each time you make changes to an app, it creates a new app version and this app version gets launched in a new environment, meaning that you could potentially switch between environments(or versions).

Regarding the second observation, yes, had a lapsus there, I'm pushing a fix.